src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -353,12 +353,13 @@ private BoundExpression RewriteStringConcatenationTwoExprs(SyntaxNode syntax, Bo
                 var readOnlySpanOfChar = readOnlySpanCtorRefParamGeneric.ContainingType.Construct(_compilation.GetSpecialType(SpecialType.System_Char));
                 var readOnlySpanCtorRefParamChar = readOnlySpanCtorRefParamGeneric.AsMember(readOnlySpanOfChar);
 
-                var locals = ArrayBuilder<LocalSymbol>.GetInstance(capacity: 2);
-                locals.AddRange(previousLocals);
+                var locals = ArrayBuilder<LocalSymbol>.GetInstance(capacity: 1);
 
                 var wrappedLeft = WrapIntoReadOnlySpanOfCharIfNecessary(loweredLeft, stringImplicitConversionToReadOnlySpan, readOnlySpanCtorRefParamChar, out var leftTempLocal);
                 var wrappedRight = WrapIntoReadOnlySpanOfCharIfNecessary(loweredRight, stringImplicitConversionToReadOnlySpan, readOnlySpanCtorRefParamChar, out var rightTempLocal);
 
+                Debug.Assert(leftTempLocal is null || rightTempLocal is null);
```

Comment:
One more small commment: I think it would also be worth asserting `leftIsChar != rightIsChar`, and explaining that `c + c` can never be on the left-most side of a binary expression that can be optimized like this, because that `c + c` will result in an `int`, not a `string`. This assert is the logical consequence of that, but the original precondition is that we never get in here with 2 `char`s.